app/decorators/service_provider_session_decorator.rb

```diff
@@ -74,7 +74,7 @@ def cancel_link_path
   attr_reader :sp, :view_context, :sp_session
 
   def request_url
-    sp_session[:request_url]
+    sp_session[:request_url] || ServiceProviderRequest.find_by(issuer: sp.issuer).url
```

Comment:
This is not unique enough, many issuers will have the same SP here, don't we need the `request_id` to identify the request uniquely?

For OIDC, two requests to the same SP could have different redirect URLs and will have different `state` params and other unique things